packages/medusa/src/services/order.ts

```diff
@@ -703,22 +686,31 @@ class OrderService extends TransactionBaseService {
 
       await Promise.all(
         [
-          cart.items.map((lineItem) => {
-            const lineItemPromises: unknown[] = [
+          cart.items.map((lineItem): unknown[] => {
+            const toReturn: unknown[] = [
               lineItemServiceTx.update(lineItem.id, { order_id: order.id }),
-              inventoryServiceTx.adjustInventory(
-                lineItem.variant_id,
-                -lineItem.quantity
-              ),
             ]
 
             if (lineItem.is_giftcard) {
-              lineItemPromises.push(
+              toReturn.push(
                 this.createGiftCardsFromLineItem_(order, lineItem, manager)
               )
             }
 
-            return lineItemPromises
+            if (lineItem.variant_id) {
+              toReturn.push(
+                this.productVariantInventoryService_.reserveQuantity(
```

Comment:
thought (non-blocking): maybe it would make sense to move this into the cart completion strategy. My thinking is that consumers of the order service may want to have different workflows depending on their needs so decoupling the "reservation" step from the "placement" step would give more optionality. Nothing to do right now, but may make sense to include when we need to make the fulfillments location aware.

Reply:
Really like this idea!

Reply:
I think this makes a lot of sense, I have moved this into the cart completion strategy 😄